li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -319,6 +322,9 @@ public void unsubscribe() {
         }
       } catch (OffsetOutOfRangeException | NoOffsetForPartitionException oe) {
         handleInvalidOffsetException(oe);
+        if (_throwExceptionOnInvalidOffsets) {
```

Comment:
No, the behavior we want is to let LiKafkaConsumerImpl to handleInvalidOffsetException (i.e. auto reset to latest/earliest/closest based on config) and also throw the exception so that XC can know an exception happened and invoke the user callback. The end goal is to move the large msg related logic to XC layer so that we only use the raw kafka consumer underlying so everything is handled at XC layer. 

This is just a temp solution since if we handleInvalidOffsetException in XC layer, we'll mess up the ConsumerRecordsProcessor state (large msg hwm, safe offset etc), so we'll let LiKafkaConsumerImpl handle it as it is now, but force it to throw exception.

Reply:
mm ok. So, on exception, xinfra client invokes user callback? dumb question: what user callback are we referring to here?

Reply:
1. Yes
2. We don't have any concrete usage yet, but likely pause/re-assign etc.